### PR TITLE
Add nightly linkchecker

### LINKED ISSUE #672: Add nightly linkchecker
We should add a nightly linkchecker to avoid broken links.

The amount of charms is still manageable to run linkchecker on the entire site.

### PR DESCRIPTION
**Pull request by solazio**
## Done
- _Add nightly linkchecker._

## Issue / Card
Fixes #672

### CODE REVIEW

File: .github/workflows/linkchecker.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+      - name: Install linkchecker
+        run: sudo pip install LinkChecker
+
+      - name: Run linkchecker
+        run: linkchecker --threads 2 --ignore-url "http://localhost*" --ignore-url "https://assets.ubuntu.com$" --check-extern --no-warnings https://charmhub.io
+
+      - name: Send message on failure
+        if: failure()
+        run: curl -X POST -F "workflow=${GITHUB_WORKFLOW}" -F "repo_name=${GITHUB_REPOSITORY}" -F "action_id=${GITHUB_RUN_ID}" ${{ secrets.BOT_URL }}?room=%23web-team
```
Comment (tbille) [inline .github/workflows/linkchecker.yaml:24]: not sure about the ping on the webteam channel

Comment (solazio) [reply .github/workflows/linkchecker.yaml:24]: @anthonydillon is using [docs](https://github.com/canonical-web-and-design/multipass.run/blob/4818bab04a16028ca67cca5367f6817a30fe73ee/.github/workflows/docs-links.yaml#L24) for Docs link check. Shall we create a separate channel for linkchecker notifications?

Comment (anthonydillon) [reply .github/workflows/linkchecker.yaml:24]: Webteam would be fine as its information we should want to know. Creating a new room is likely for these notifications to be missed,

File: .github/workflows/linkchecker.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    steps:
+      - name: Checkout master
+      - uses: actions/checkout@v2
+
+      - name: Install linkchecker
+        run: sudo pip install LinkChecker
+
+      - name: Run linkchecker
+        run: linkchecker --threads 2 --ignore-url "http://localhost*" --ignore-url "https://assets.ubuntu.com$" --check-extern --no-warnings https://charmhub.io
```
Comment (anthonydillon) [inline .github/workflows/linkchecker.yaml:20]: You could remove the `--threads 2` so it defaults to 10 to speed up the checks

File: .github/workflows/linkchecker.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: Run linkchecker on site
+
+on:
+  schedule:
+    - cron:  '0 0 1,15 * *'
```
Comment (anthonydillon) [inline .github/workflows/linkchecker.yaml:5]: 00:00 UTC is a strange time to run. Probably more efficient to run in main working hours.

**Review by anthonydillon — COMMENTED** [review]
This is not set to nightly.

File: .github/workflows/linkchecker.yaml (hunk 3 above)
Comment (solazio) [reply .github/workflows/linkchecker.yaml:5]: fair enough, I'll run it at 12:00

File: .github/workflows/linkchecker.yaml (hunk 1 above)
Comment (solazio) [reply .github/workflows/linkchecker.yaml:24]: In that case I keep it as it is.

Comment (tbille) [reply .github/workflows/linkchecker.yaml:24]: with the `--check-extern` if any of the charms publisher posts a 404 links we will always get this message until the publisher fixes it

**Comment by solazio** [comment]
> This is not set to nightly.

@anthonydillon is set to run on the 1st and 15th of every month... because I definitely had in mind that it has to run fortnightly. I will change it to run every day.

**Comment by anthonydillon** [comment]
> @anthonydillon is set to run on the 1st and 15th of every month... because I definitely had in mind that it has to run fortnightly. I will change it to run every day.

I don't mind which you use but the code is fortnightly and the commit says nightly.

**Review by tbille — APPROVED** [review]
as discussed in standup let's see how it goes.

**Pull request merged**